assets/l10n/app_en.arb

```diff
@@ -289,6 +289,21 @@
   "@loginServerUrlInputLabel": {
     "description": "Input label in login page for Zulip server URL entry."
   },
+  "serverURLDocLinkLabel": "What's this?",
```

Comment:
nit: In camel case, capitalize acronyms like "URL" the same way as words:
```suggestion
  "serverUrlDocLinkLabel": "What's this?",
```

See the Dart style guide for discussion: https://dart.dev/effective-dart/style#do-capitalize-acronyms-and-abbreviations-longer-than-two-letters-like-words